app/jobs/ingest_file_job.rb

```diff
@@ -2,36 +2,42 @@ class IngestFileJob < ActiveJob::Base
   queue_as CurationConcerns.config.ingest_queue_name
 
   # @param [FileSet] file_set
-  # @param [String] filename the cached file within the CurationConcerns.config.working_path
-  # @param [String,NilClass] mime_type
+  # @param [String] filepath the cached file within the CurationConcerns.config.working_path
   # @param [User] user
-  # @param [String] relation ('original_file')
-  def perform(file_set, filename, mime_type, user, relation = 'original_file')
-    local_file = File.open(filename, "rb")
+  # @option opts [String] mime_type
+  # @option opts [String] filename
+  # @option opts [String] relation, ex. :original_file
+  def perform(file_set, filepath, user, opts = {})
+    mime_type = opts.fetch(:mime_type, nil)
+    filename = opts.fetch(:filename, File.basename(filepath))
+    relation = opts.fetch(:relation, :original_file).to_sym
```

Comment:
Is `#to_sym` required? Would `#to_s` suffice? Creating symbols prior to Ruby 2.2 (or so) means creating objects that are not cleaned up.